libvirt/domain.go

```diff
@@ -230,6 +235,134 @@ func newDiskForCloudInit(virConn *libvirt.Connect, volumeKey string) (libvirtxml
 	return disk, nil
 }
 
+func getFirstDiskPath(d *schema.ResourceData, domainDef *libvirtxml.Domain, virConn *libvirt.Connect) (string, error) {
+	prefix := fmt.Sprintf("disk.%d", 0)
+
+	if volumeKey, ok := d.GetOk(prefix + ".volume_id"); ok {
+		diskVolume, err := virConn.LookupStorageVolByKey(volumeKey.(string))
+		if err != nil {
+			return "", fmt.Errorf("Error lookup storage disk volume: %s", err)
+		}
+		diskVolumeFile, err := diskVolume.GetPath()
+		if err != nil {
+			return "", fmt.Errorf("Error get path of disk volume: %s", err)
+		}
+		return diskVolumeFile, nil
+	}
+	return "", fmt.Errorf("failed to find disk 0")
+}
+
+func getStderr(stderr io.ReadCloser) string {
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(stderr)
+	return buf.String()
+}
+
+func inlineupdateCoreOSIgnition(d *schema.ResourceData, domainDef *libvirtxml.Domain, virConn *libvirt.Connect) error {
+	if ignition, ok := d.GetOk("coreos_ignition"); ok {
+		//basically we are doing following scripts
+		//eval  $(guestfish --listen -a $1)
+		//guestfish --remote -- run
+		//guestfish --remote -- findfs-label
+		//guestfish --remote -- mount $2 /
+		//guestfish --remote -- mkdir-p $3
+		//guestfish --remote -- upload $4 $3/$4
+		//guestfish --remote -- umount-all
+		//guestfish --remote -- exit
+
+		// Get the root disk (first disk) volume location
+		// FIXME: currently assume first one will be the boot disk
```

Comment:
Is better to have a Github ISSUE and removing the `fixme` if we can't fix this now

Reply:
I will open an issue after this PR merged